tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTSecurityCRLTest.java

```diff
@@ -0,0 +1,357 @@
+
+      ActiveMQJAASSecurityManager securityManager = (ActiveMQJAASSecurityManager) server.getSecurityManager();
+
+      securityManager.getConfiguration().addUser(fullUser, fullPass);
+      securityManager.getConfiguration().addRole(fullUser, "full");
+      // Configure roles
+      HierarchicalRepository<Set<Role>> securityRepository = server.getSecurityRepository();
+      HashSet<Role> value = new HashSet<>();
+      value.add(new Role("nothing", false, false, false, false, false, false, false, false));
+      value.add(new Role("browser", false, false, false, false, false, false, false, true));
+      value.add(new Role("guest", false, true, false, false, false, false, false, true));
+      value.add(new Role("full", true, true, true, true, true, true, true, true));
+      securityRepository.addMatch("#", value);
+
+      server.getConfiguration().setSecurityEnabled(true);
+   }
+
+   public static TrustManager[] getTrustManager(String truststorePath, String truststorePass) throws Exception {
+      KeyStore trustedCertStore = KeyStore.getInstance("jks");
+
+      trustedCertStore.load(new FileInputStream(truststorePath), truststorePass.toCharArray());
+      TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+
+      tmf.init(trustedCertStore);
+      return tmf.getTrustManagers();
+   }
+
+   public static KeyManager[] getKeyManager(String keystorePath, String keystorePass) throws Exception {
+      KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+      KeyStore ks = KeyStore.getInstance("jks");
+      KeyManager[] keystoreManagers = null;
+
+      byte[] sslCert = loadClientCredential(keystorePath);
+
+      if (sslCert != null && sslCert.length > 0) {
+         ByteArrayInputStream bin = new ByteArrayInputStream(sslCert);
+         ks.load(bin, keystorePass.toCharArray());
+         kmf.init(ks, keystorePass.toCharArray());
+         keystoreManagers = kmf.getKeyManagers();
+      }
+      return keystoreManagers;
+   }
+
+   private static byte[] loadClientCredential(String fileName) throws IOException {
+      if (fileName == null) {
+         return null;
+      }
+      FileInputStream in = new FileInputStream(fileName);
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      byte[] buf = new byte[512];
+      int i = in.read(buf);
+      while (i > 0) {
+         out.write(buf, 0, i);
+         i = in.read(buf);
+      }
+      in.close();
+      return out.toByteArray();
+   }
+
+   private BlockingConnection retrieveMQTTConnection(String host, String truststorePath, String truststorePass, String keystorePath, String keystorePass) throws Exception {
+
+      MQTT mqtt = new MQTT();
+      mqtt.setConnectAttemptsMax(1);
+      mqtt.setReconnectAttemptsMax(0);
+      mqtt.setHost(host);
+      mqtt.setUserName(fullUser);
+      mqtt.setPassword(fullPass);
+      SSLContext sslContext = SSLContext.getInstance("TLS");
```

Comment:
Why not use the SSLSupport class that's already included in the code-base to generate the SSLContext?